frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -236,8 +242,19 @@ pub mod pallet {
 
     #[pallet::hooks]
     impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
-        fn on_initialize(_now: BlockNumberFor<T>) -> Weight {
+        fn on_initialize(now: BlockNumberFor<T>) -> Weight {
             // just return the weight of the on_finalize.
+            let force_new_era = Self::force_era().eq(&Forcing::ForceNew);
+            if (now % T::BlockPerEra::get()).is_zero() || force_new_era {
+                Self::distribute_era_rewards();
```

Comment:
on_initialize method has very limited execution time, usually 10% of block, please be sure that all your code has constant execution time.

Reply:
thanks. Let's take the balance snapshot.